Frends.ServiceBus.Read/Frends.ServiceBus.Read/Read.cs

```diff
@@ -0,0 +1,271 @@
+            }
+            await manager.CreateQueueAsync(queueDescription, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    private static async Task EnsureTopicExists(string connectionString, string queueOrTopicName, string subscriptionName, TimeSpan deleteIdle, int maxSize, CancellationToken cancellationToken)
+    {
+        var managementClient = new ManagementClient(connectionString);
+        TopicDescription topicDescription;
+
+        if (!await managementClient.TopicExistsAsync(queueOrTopicName, cancellationToken).ConfigureAwait(false))
+        {
+            if (deleteIdle == TimeSpan.Zero)
+            {
+                topicDescription = new TopicDescription(queueOrTopicName)
+                {
+                    EnableBatchedOperations = true,
+                    MaxSizeInMB = maxSize,
+                };
+            }
+            else
+            {
+                topicDescription = new TopicDescription(queueOrTopicName)
+                {
+                    EnableBatchedOperations = true,
+                    MaxSizeInMB = maxSize,
+                    AutoDeleteOnIdle = deleteIdle,
+                };
+            }
+            await managementClient.CreateTopicAsync(topicDescription, cancellationToken).ConfigureAwait(false);
+        }
+        await EnsureSubscriptionExists(queueOrTopicName, subscriptionName, connectionString, deleteIdle, cancellationToken);
+    }
+
+    private static async Task EnsureSubscriptionExists(string queueOrTopicName, string subscriptionName, string connectionString, TimeSpan deleteIdle, CancellationToken cancellationToken)
+    {
+        var manager = new ManagementClient(connectionString);
+        SubscriptionDescription subscriptionDescription;
+
+        if (!await manager.SubscriptionExistsAsync(queueOrTopicName, subscriptionName, cancellationToken).ConfigureAwait(false))
+        {
+            if (deleteIdle == TimeSpan.Zero)
+            {
+                subscriptionDescription = new SubscriptionDescription(queueOrTopicName, subscriptionName)
+                {
+                    EnableBatchedOperations = true,
+                };
+            }
+            else
+            {
+                subscriptionDescription = new SubscriptionDescription(queueOrTopicName, subscriptionName)
+                {
+                    EnableBatchedOperations = true,
+                    AutoDeleteOnIdle = deleteIdle,
+                };
+            }
+            await manager.CreateSubscriptionAsync(subscriptionDescription, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+
+    private static async Task<ReadResult> DoReadOperation(Input input, Options options, CancellationToken cancellationToken)
+    {
+        ServiceBusConnection connection = null;
+        MessageReceiver requestClient = null;
+
+        var path = input.SourceType == QueueOrTopic.Topic ? $"{input.QueueOrTopicName}/subscriptions/{input.SubscriptionName}" : input.QueueOrTopicName;
```

Comment:
IMO this should also be put to multiple lines.

Reply:
I think it's cleaner to keep these in 1 line because you can word wrap these lines in Visual Studio when using smaller resolution.